keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -123,21 +123,45 @@ class ImageService(
    * Returns the latest image that is present in all regions.
    * Each ami must have tags.
    */
-  suspend fun getLatestNamedImageWithAllRegions(packageName: String, account: String, regions: List<String>): NamedImage? =
-    cloudDriverService.namedImages(
+  suspend fun getLatestNamedImageWithAllRegions(packageName: String, account: String, regions: List<String>): NamedImage? {
+    val images = cloudDriverService.namedImages(
       serviceAccount = DEFAULT_SERVICE_ACCOUNT,
       imageName = packageName,
       account = account
     )
+
+    val filteredImages = images
       .filter { it.hasAppVersion }
       .sortedWith(NamedImageComparator)
+
+    val eliminatedImages = mutableMapOf<String, String>()
+    val image = filteredImages
       .find {
+        var errors = ""
         val curAppVersion = AppVersion.parseName(it.appVersion)
-        curAppVersion.packageName == packageName &&
-          it.accounts.contains(account) &&
-          it.amis.keys.containsAll(regions) &&
-          tagsExistForAllAmis(it.tagsByImageId)
+        if (curAppVersion.packageName != packageName) {
+          errors += "[package name ${curAppVersion.packageName} does not match required package]"
```

Comment:
Would it be best perhaps to make this a list and add each message as an item and then `.join("\n")` at the end so that they're not all in one line?

Reply:
😆 probably. I'll do that